pkg/cluster/cluster.go

```diff
@@ -867,41 +867,52 @@ func (c *Cluster) GetStatus() *spec.ClusterStatus {
 }
 
 // Switchover does a switchover (via Patroni) to a candidate pod
-func (c *Cluster) Switchover(curMaster *v1.Pod, candidate spec.NamespacedName) error {
+func (c *Cluster) Switchover(curMaster *v1.Pod, candidate spec.NamespacedName) (err error) {
+
 	c.logger.Debugf("failing over from %q to %q", curMaster.Name, candidate)
 
+	var wg sync.WaitGroup
+
 	podLabelErr := make(chan error)
 	stopCh := make(chan struct{})
-	defer close(podLabelErr)
+
+	wg.Add(1)
 
 	go func() {
 		ch := c.registerPodSubscriber(candidate)
+		defer wg.Done()
```

Comment:
Should this be below the next defer just to make sure that it unblocks _after_ it unregistered pod subscriber? Not sure if ordering is an issue here, just thought there might be a possibility

Reply:
Moving wg.Done() to the top of this function makes sense, although the defer order is correct, since they are pushed to the stack and then unrolled in the last-in-first-out order.

Reply:
that's right, my mistake :D